docs/contribute.rst

```diff
@@ -74,57 +44,99 @@ as something that could/will be pasted into release notes:
 Why should I care?
 
 * It helps others (and yourself) find relevant commits quickly.
-* The summary line can be re-used later (e.g. for rpm changelog).
+* The summary line will be re-used later (e.g. for rpm changelog).
 * Some tools do not handle wrapping, so it is then hard to read.
-* You will make the maintainer happy to read beautiful commits :)
+* You will make the maintainers happy to read beautiful commits :)
 
 You can get some more context in the `stackoverflow`__ article.
 
 __ http://stackoverflow.com/questions/2290016/
 
 
+Develop
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In order to experiment, play with the latest bits and develop
+improvements it is best to use a virtual environment::
+
+    mkvirtualenv tmt
+    git clone https://github.com/psss/tmt
+    cd tmt
+    pip install -e .
+
+Install ``python3-virtualenvwrapper`` to easily create and enable
+virtual environments using ``mkvirtualenv`` and ``workon``. Note
+that if you have freshly installed the package you need to open a
+new shell session to enable the wrapper functions.
+
+The main ``tmt`` package contains only the core dependencies. For
+building documentation, testing changes, importing/exporting test
+cases or advanced provisioning options install the extra deps::
+
+    pip install '.[docs]'
+    pip install '.[tests]'
+    pip install '.[convert]'
+    pip install '.[provision]'
+
+Or simply install all extra dependencies to make sure you have
+everything needed for the tmt development ready on your system::
+
+    pip install '.[all]'
+
+
 Tests
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+Every code change should be accompanied by tests covering the new
+feature or affected code area. It's possible to write new tests or
+extend the existing ones. If writing a test is not feasible for
+you, explain the reason in the pull request. If possible, the
+maintainers will help with creating needed test coverage.
```

Comment:
Didn't we discuss some time ago that adding a label meaning "help with tests wanted" would be a good idea? I don't completely remember what was the outcome but it would be worth mentioning if we came to a conclusion

Reply:
Thanks for reminding this. From the notes it seems we agred on a new `tests needed` label. I've just created it and going to add it to the doc.